packages/web3-providers/src/magiceden/index.ts

```diff
@@ -0,0 +1,209 @@
+import urlcat from 'urlcat'
+import { first } from 'lodash-unified'
+import { ChainId, FungibleTokenDetailed } from '@masknet/web3-shared-evm'
+import type { MagicEdenTokenResponse } from './types'
+import { MAGICEDEN_MAINNET_API_URL } from './constants'
+import { NonFungibleTokenAPI } from '..'
+import { isProxyENV } from '../helpers'
+
+async function fetchFromMagicEden<T>(url: string, path: string, init?: RequestInit) {
```

Comment:
```suggestion
async function fetchFromMagicEden(url: string, path: string, init?: RequestInit): Promise<unknown> {
```

You should do a structure check on the returned value instead of casting it via generics.